internal/workflow.go

```diff
@@ -46,92 +46,229 @@ var (
 )
 
 type (
-	// Channel must be used instead of native go channel by workflow code.
-	// Use workflow.NewChannel(ctx) method to create Channel instance.
+	// Channel must be used in workflows instead of a native Go chan.
+	//
+	// Use workflow.NewChannel(ctx) to create an unbuffered Channel instance,
+	// workflow.NewBufferedChannel(ctx, size) to create a Channel which has a buffer,
+	// or workflow.GetSignalChannel(ctx, "name") to get a Channel that can contain encoded data sent from other systems.
```

Comment:
`encoded data sent from other systems.` is a little confusing to me. 

I think we just need to mention this is a special channel that maintained/created by Cadence just for workflow signal feature.

Reply:
yeah, I can reword this.
the main difference is that the internal channels _do not_ encode their data, so they can pass non-serializable data around.  it's a somewhat significant difference, IMO it needs to be covered somewhere.

but it doesn't have to be here / in this way :)  lemme think about it / I'm open to suggestions!

Reply:
yeah I think just `other systems` bugs me a little. Without context I have no clue what is `other` mean here. I think just say `contain encoded data sent by workflow signal` should be good for me.

Good point about the data encoding. May worth mentioning other channels are `workflow internal channels so that they can pass in/out variables that are not serializable`